src/content/docs/en/basics/astro-pages.mdx

```diff
@@ -96,10 +96,52 @@ Files with the `.html` file extension can be placed in the `src/pages/` director
 
 ## Custom 404 Error Page
 
-For a custom 404 error page, you can create a `404.astro` or `404.md` file in `/src/pages`.
+For a custom 404 error page, you can create a `404.astro` or `404.md` file in `src/pages`.
 
 This will build to a `404.html` page. Most [deploy services](/en/guides/deploy/) will find and use it.
 
+## Custom 500 Error Page
+
+For a custom 500 error page to show for pages in your `hybrid` or `server` project that are [rendered on demand](/en/guides/server-side-rendering/#enable-on-demand-server-rendering), create the file `src/pages/500.astro`. This custom page is not available for prerendered pages.
+
+If an error occurs rendering this page, your host's default 500 error page will be shown to your visitor.
+
+### Use the `error` prop
+
+<p><Since v="4.10.0" /></p>
+
+`500.astro` accepts an `error` prop. It can be anything, so make sure to handle it properly:
```

Comment:
```suggestion
`src/pages/500.astro` is a special page that is automatically passed an `error` prop for any error thrown during rendering. This allows you to use the details of an error (e.g. from a page, from middleware, etc.) to display information to your visitor. 

When using TypeScript, you will need to account for the fact that the `error` prop's data type cannot be determined ahead of time:
```
I don't know about that second paragraph, but like, whatever is true, maybe something like that?

Reply:
I updated your suggestion (2nd paragraph) wdyt? It's important (I think) to note that is not a type-level only thing, it's also a runtime thing. I'm afraid talking about TS here would make JS only people skip that part

Reply:
To be clear, if someone were not writing TypeScript, would they need to specify the props (or, simply write their logic to account for an unknown data type)?

I think that's what I'd want to be clear about. "You need to account for the data type" still doesn't scream "think about the run time!" to me, and might make people instead think only that they need to type the props.

Reply:
Someone not using TS would not need to define the interface but still need to handle the error in the template. I like the suggestion, happy with whatever wording you prefer!

Reply:
I'm happy enough with my suggestion for now, too! (We have a whole week before minor release!) :joy: 

Let's go with that, and I'll consider this good to go unless I have last minute thoughts!